pyvista/plotting/plotting.py

```diff
@@ -4203,21 +4226,18 @@ def show(self, title=None, window_size=None, interactive=True,
         #       See issues #135 and #186 for insight before editing the
         #       remainder of this function.
 
-        # Get camera position before closing
-        cpos = self.camera_position
-
-        # Cleanup
+        # Close the render window if requested
         if auto_close:
             self.close()
 
         # If user asked for screenshot, return as numpy array after camera
         # position
         if return_img or screenshot is True:
-            return cpos, self.last_image
+            if return_cpos:
+                return self.camera_position, self.last_image
 
-        # Return last used camera position unless within a doctest
-        if '_pytest.doctest' not in sys.modules:
-            return cpos
+        if return_cpos:
+            return self.camera_position
```

Comment:
I know the signature is getting messy. But where's the `widget` being returned when asked for?

Reply:
https://github.com/pyvista/pyvista/blob/ab3b37d8ebab175fce83cf199a352e160892bbc8/pyvista/plotting/plotting.py#L4159-L4168

Reply:
Thanks. So this looks like when `return_viewer=True` and there's a jupyter backend, nothing else is returned, which is unintuitive and should be made clear: (return jupyter viewer) XOR (return cpos and/or image array depending on flags, in this order).

Reply:
Is this sufficient? c05322dd2db6bd175e2f0256909501bd2cba82bc